Poisson_surface_reconstruction_3/include/CGAL/Poisson_mesh_domain_3.h

```diff
@@ -0,0 +1,427 @@
+#ifndef DOXYGEN_RUNNING
+    , typename std::enable_if<!is_named_function_parameter<Function>>::type* = nullptr
+#endif // DOXYGEN_RUNNING
+  )
+    : Base(make_implicit_to_labeling_function_wrapper<BGT>(function), bounding_object, np),
+    poisson_function(function)
+  {}
+
+  /*!  \brief Construction from a function, a bounding object and a relative error bound.
+   *
+   * \tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+   *
+   * \param function the Poisson reconstruction function
+   * \param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below:
+   *
+   * \cgalNamedParamsBegin
+   *   \cgalParamNBegin{relative_error_bound}
+   *      \cgalParamDescription{the relative error bound used to compute intersection points between the implicit surface and query segments.
+   *                            The bisection is stopped when the length of the intersected segment is less than the product
+   *                            of `relative_error_bound` by the diameter of the bounding object.}
+   *      \cgalParamDefault{FT(1e-3)}
+   *   \cgalParamNEnd
+   * \cgalNamedParamsEnd
+   */
+  template<typename CGAL_NP_TEMPLATE_PARAMETERS>
+  Poisson_mesh_domain_3(const Function & function,
+    const CGAL_NP_CLASS& np = parameters::default_values()
+#ifndef DOXYGEN_RUNNING
+    , typename std::enable_if<!is_named_function_parameter<Function>>::type * = nullptr
+#endif // DOXYGEN_RUNNING
+  )
+    : Base(make_implicit_to_labeling_function_wrapper<BGT>(function), function.bounding_sphere(), np),
+    poisson_function(function)
+  {}
+///@}
+
+#ifndef DOXYGEN_RUNNING
+  template<typename CGAL_NP_TEMPLATE_PARAMETERS_NO_DEFAULT>
+  Poisson_mesh_domain_3(const CGAL_NP_CLASS& np)
+      : Base(np)
+  {}
+
+  // Overload handling parameters passed with operator=
+
+  template<typename CGAL_NP_TEMPLATE_PARAMETERS_NO_DEFAULT_1,
+           typename CGAL_NP_TEMPLATE_PARAMETERS_NO_DEFAULT_2,
+           typename ... NP>
+  Poisson_mesh_domain_3(const Function& function,
+                        const CGAL_NP_CLASS_1& np1,
+                        const CGAL_NP_CLASS_2&  np2,
+                        const NP& ... nps)
+      : Base(internal_np::combine_named_parameters(
+            CGAL::parameters::function(make_implicit_to_labeling_function_wrapper<BGT>(function)), np1, np2, nps...)),
+        poisson_function(function)
+  {}
+#endif
+
+/// \name Creation of domains from Poisson implicit functions
+/// @{
+  /*! \brief Construction from a Poisson implicit function
+   *
+   * This static method is a <em>named constructor</em>. It constructs a domain
+   * whose bounding surface is described implicitly as the zero level set of a
+   * function.  The domain to be discretized is assumed to be the domain where
+   * the function has negative values.
+   *
+   * The method takes as argument a bounding sphere which is required to
```

Comment:
this takes more than a sphere (probably the same fix is needed in the base class)